app/services/core_data_connector/authority/bnf.rb

```diff
@@ -0,0 +1,48 @@
+require 'json'
+require 'active_support/core_ext'
+require 'typhoeus'
+
+module CoreDataConnector
+  module Authority
+    class Bnf
+      BASE_URL = 'https://catalogue.bnf.fr/api/SRU'
+
+      DEFAULT_LIMIT = 20
+
+      BASE_PARAMS = {
+        operation: 'searchRetrieve',
+        startRecord: 1,
+        recordSchema: 'dublincore',
+        version: 1.2
+      }
+
+      def find(id, options = {})
+        params = {
+          query: "bib.ark+all+\"#{id}\"",
+          maximumRecords: options[:limit] || DEFAULT_LIMIT
+        }.merge(BASE_PARAMS)
+
+        base_request(id, params)
+      end
+
+      def search(query, options = {})
+        params = {
+          query: "bib.anywhere+all+\"#{query}\"",
+          maximumRecords: options[:limit] || DEFAULT_LIMIT
+        }.merge(BASE_PARAMS)
+
+        base_request(query, params)
+      end
+
+      protected
+
+      def base_request(query, params)
+        request = Typhoeus::Request.new(BASE_URL, method: :get, params: params)
```

Comment:
My hope was that the `http` module would include all of the HTTP calls and error handling. Could we refactor that module to be able to handle the data from responses in different ways? Perhaps using blocks?

```
send_request(url) do |response|
  JSON.parse(response.body)
end

send_request(url) do |response|
  Hash.from_xml(response.body).to_json
end
```